### DISCUSSION TITLE
Why was native validator support removed?

### DISCUSSION THREAD

**Discussion by onionhammer** [react-hook-form/react-hook-form#1985] [General] [answered]
Seems like this lowers the amount of effort to create a rich form with validation; why remove the option?

**Comment by bluebill1049** [comment, marked as answer]
1. the usage is very low
2. save package size

**Reply by bluebill1049** [reply]
I am happy to bring native validation back which allow users to set custom error message, but i think the demand needs to be there otherwise the extra code is a burden for the rest of the community. I wouldn't mind to see a feature request and tagged it with waiting for up vote.

**Reply by onionhammer** [reply]
Would it be possible to be removed with dead code elimination of it's not used?

**Reply by bluebill1049** [reply]
yea, tree shake is not going to be a good friend here, because the core will be injected into the core.
